controller/backing_image_manager_controller.go

```diff
@@ -588,9 +588,38 @@ func (c *BackingImageManagerController) deleteInvalidBackingImages(bim *longhorn
 			}
 			return err
 		}
-		if bi.Status.UUID == bim.Spec.BackingImages[biName] {
+
+		// Delete the file from a backing image manager when:
+		//   1. The spec record is removed or does not match.
+		//   2. The file state is failed and there are available files in
+		//      other backing image managers.
+		deleteRequired := false
+		if bi.Status.UUID != bim.Spec.BackingImages[biName] {
+			deleteRequired = true
+		}
+		if !deleteRequired {
+			// Prefer to use the file info in BackingImage.Status
+			fileState := bim.Status.BackingImageFileMap[biName].State
+			if bi.Status.DiskFileStatusMap[bim.Spec.DiskUUID] != nil {
+				fileState = bi.Status.DiskFileStatusMap[bim.Spec.DiskUUID].State
+			}
```

Comment:
What's difference between the two `fileState`?

Reply:
BackingImageController will sync the file status from the BIM CRs to the BI CRs but with some adjustments. The final status will be recorded/summarized in BI CRs. e.g.,:
https://github.com/longhorn/longhorn-manager/blob/167888d93bcb763d9cdff0d02b8c02f236912a33/controller/backing_image_controller.go#L716-L729

Reply:
Got it, thank you.